### PR TITLE
VB-1926 Update orchestration service to acquire full username and ret…

### PR DESCRIPTION
**Pull request by AledWEvans**
This code completes the VB-1926 in order to return the correct information to the front end. 
It also makes sure we call Auth as few times as possible.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt [hunk 1]
```diff
@@ -3,10 +3,12 @@ package uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.client
...
 import org.springframework.stereotype.Component
 import reactor.core.publisher.Mono
-import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.hmpps.auth.UserDetails
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.hmpps.auth.UserDetailsDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.orchestration.VisitHistoryDetailsDto
 import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.visit.scheduler.VisitDto
 import java.time.Duration
-import java.util.Optional
+import java.time.LocalDateTime
+import java.util.*
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt:11]: not sure how the lint checks didn't pick this.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt [hunk 2]
```diff
@@ -17,37 +19,90 @@ class VisitDetailsClient(
...
+    return results.toMap()
   }
 
-  private fun getFullName(userDetails: Optional<UserDetails>?): String? {
-    return if (userDetails?.isEmpty == true) null else userDetails?.get()?.name
+  private fun createUserMonoCalls(
+    visitDto: VisitDto,
+  ): List<Mono<UserDetailsDto>> {
+    val userNames = mutableSetOf(visitDto.createdBy, visitDto.updatedBy, visitDto.cancelledBy)
+    userNames.remove(null)
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt:94]: minor comment - we could combine this on Line 93. Trivial, can be ignored.

Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt:94]: `public abstract fun remove(
    element: E
): Boolean`

remove is not builder pattern will see if I can find something else.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt (hunk 1 above)
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt:11]: You mean the .* I don't actually think it should be checking for such things

Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt:11]: ok.. I thought it fails lint check if we do a *. But yes can mark it resolved for now. Can change if it gets picked at a alter stage.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt (hunk 2 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/client/VisitDetailsClient.kt]: Ignore. I thought it might be.

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**